src/operations/delete.ts

```diff
@@ -105,22 +108,18 @@ export class DeleteOneOperation extends DeleteOperation {
     super(collection.s.namespace, [makeDeleteStatement(filter, { ...options, limit: 1 })], options);
   }
 
-  override executeCallback(
+  override async execute(
     server: Server,
-    session: ClientSession | undefined,
-    callback: Callback<DeleteResult>
-  ): void {
-    super.executeCallback(server, session, (err, res) => {
-      if (err || res == null) return callback(err);
-      if (res.code) return callback(new MongoServerError(res));
-      if (res.writeErrors) return callback(new MongoServerError(res.writeErrors[0]));
-      if (this.explain) return callback(undefined, res);
-
-      callback(undefined, {
-        acknowledged: this.writeConcern?.w !== 0 ?? true,
-        deletedCount: res.n
-      });
-    });
+    session: ClientSession | undefined
+  ): Promise<DeleteResult> {
+    const res = await super.execute(server, session);
+    if (res.code) throw new MongoServerError(res);
+    if (res.writeErrors) throw new MongoServerError(res.writeErrors[0]);
+    if (this.explain) return res;
+    return {
```

Comment:
Can we move the logic from this `execute` and `DeleteManyOperation`'s execute into the `DeleteOperation`?  Then these classes don't need `execute` methods at all.

Reply:
how do we do that

Reply:
We should be able to move the logic right after the call to `super.executeCommand()` in delete operation, right?

Reply:
in delete operation, it returns super.executeCommand(), so it would never reach the logic

Reply:
```typescript
export class DeleteOperation extends CommandOperation<DeleteResult> {
...
  override async execute(
    server: Server,
    session: ClientSession | undefined
  ): Promise<DeleteResult> {
    const options = this.options ?? {};
    const ordered = typeof options.ordered === 'boolean' ? options.ordered : true;
    const command: Document = {
      delete: this.ns.collection,
      deletes: this.statements,
      ordered
    };
    if (options.let) {
      command.let = options.let;
    }
    // we check for undefined specifically here to allow falsy values
    // eslint-disable-next-line no-restricted-syntax
    if (options.comment !== undefined) {
      command.comment = options.comment;
    }
    const unacknowledgedWrite = this.writeConcern && this.writeConcern.w === 0;
    if (unacknowledgedWrite) {
      if (this.statements.find((o: Document) => o.hint)) {
        // TODO(NODE-3541): fix error for hint with unacknowledged writes
        throw new MongoCompatibilityError(`hint is not supported with unacknowledged writes`);
      }
    }

    const res = super.executeCommand(server, session, command) as TODO_NODE_3286;
    // new logic here
    if (res.code) throw new MongoServerError(res);
    if (res.writeErrors) throw new MongoServerError(res.writeErrors[0]);
    if (this.explain) return res;

    return {
      acknowledged: this.writeConcern?.w !== 0,
      deletedCount: res.n
    };
  }
}
```
^ something like this?

Reply:
got it. would there just be no execute function for delete one and delete many?

Reply:
That's my thought.  You can double check that the logic is the same in `DeleteOne.execute` and `DeleteMany.execute` by copy-pasting one into the other and confirming there are no changes with Git.

This works because `DeleteOperation` defines the `execute` method, and then each specialized class just inherits it.